scripts/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_presenter.py

```diff
@@ -0,0 +1,171 @@
+        table = self._view.get_table_contents()
+        self._model.clear_pairs()
+        for entry in table:
+            pair = MuonPair(pair_name=str(entry[0]),
+                            backward_group_name=str(entry[1]),
+                            forward_group_name=str(entry[2]),
+                            alpha=float(entry[3]))
+            self._model.add_pair(pair)
+
+    def update_view_from_model(self):
+        self._view.disable_updates()
+
+        self._view.clear()
+        for pair in self._model.pairs:
+            self.add_pair_to_view(pair)
+
+        self._view.enable_updates()
+
+    def update_group_selections(self):
+        groups = self._model.group_names
+        self._view.update_group_selections(groups)
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Add / Remove pairs
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def add_pair(self, pair):
+        """Add a pair to the model and view"""
+        if self._view.num_rows() > 19:
+            self._view.warning_popup("Cannot add more than 20 pairs.")
+            return
+        self.add_pair_to_view(pair)
+        self.add_pair_to_model(pair)
+
+    def add_pair_to_model(self, pair):
+        self._model.add_pair(pair)
+
+    def add_pair_to_view(self, pair):
+        self._view.disable_updates()
+        self.update_group_selections()
+        assert isinstance(pair, MuonPair)
+        entry = [str(pair.name), str(pair.forward_group), str(pair.backward_group), str(pair.alpha)]
+        self._view.add_entry_to_table(entry)
+        self._view.enable_updates()
+
+    def handle_add_pair_button_clicked(self):
+        pair = self._model.construct_empty_pair(self._view.num_rows() + 1)
+        self.add_pair(pair)
+        self.notify_data_changed()
+
+    def handle_remove_pair_button_clicked(self):
+        pair_names = self._view.get_selected_pair_names()
+        if not pair_names:
+            self.remove_last_row_in_view_and_model()
+        else:
+            self._view.remove_selected_pairs()
+            self._model.remove_pairs_by_name(pair_names)
+        self.notify_data_changed()
+
+    def remove_last_row_in_view_and_model(self):
+        if self._view.num_rows() > 0:
+            name = self._view.get_table_contents()[-1][0]
+            self._view.remove_last_row()
+            self._model.remove_pairs_by_name([name])
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Table entry validation
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def _is_edited_name_duplicated(self, new_name):
+        is_name_column_being_edited = self._view.pairing_table.currentColumn() == 0
+        is_name_unique = (sum(
```

Comment:
I expect this to be a bool and not an int

Reply:
I don't quite follow this comment, which variable is a int which you would expect to be a bool?